packages/cardpay-sdk/sdk/token-bridge-home-side.ts

```diff
@@ -47,39 +47,53 @@ const bridgedTokensQuery = `
 export default class TokenBridgeHomeSide implements ITokenBridgeHomeSide {
   constructor(private layer2Web3: Web3) {}
 
+  async relayTokens(txnHash: string): Promise<TransactionReceipt>;
   async relayTokens(
     safeAddress: string,
     tokenAddress: string,
     recipientAddress: string,
     amount: string,
     txnOptions?: TransactionOptions,
     contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt>;
+  async relayTokens(
+    safeAddressOrTxnHash: string,
+    tokenAddress?: string,
+    recipientAddress?: string,
+    amount?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
   ): Promise<TransactionReceipt> {
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
+    if (!tokenAddress) {
+      let txnHash = safeAddressOrTxnHash;
+      return await waitUntilTransactionMined(this.layer2Web3, txnHash);
+    }
+    let safeAddress = safeAddressOrTxnHash;
+
     let from = contractOptions?.from ?? (await this.layer2Web3.eth.getAccounts())[0];
     let homeBridgeAddress = await getAddress('homeBridge', this.layer2Web3);
     let token = new this.layer2Web3.eth.Contract(ERC677ABI as AbiItem[], tokenAddress);
     let symbol = await token.methods.symbol().call();
     let safeBalance = toBN(await token.methods.balanceOf(safeAddress).call());
-    if (safeBalance.lt(toBN(amount))) {
+    if (safeBalance.lt(toBN(amount!))) {
```

Comment:
ditto with the comments above around using a test to assert this exists instead of using !'s